submission/ml_pool.py

```diff
@@ -49,16 +50,6 @@ def __init__(self, base_model_pool, model_layers):
                 continue
             layers = model_layers[basemodel_identifier]
 
-            for identifier, activations_model in Hooks().iterate_hooks(basemodel_identifier, activations_model):
-                if identifier in self:  # already pre-defined
-                    continue
-                from model_tools.brain_transformation import ModelCommitment
-                # enforce early parameter binding: https://stackoverflow.com/a/3431699/2225200
-                def load(identifier=identifier, activations_model=activations_model, layers=layers):
-                    brain_model = ModelCommitment(identifier=identifier, activations_model=activations_model,
-                                                  layers=layers)
-                    for region in regions:
-                        brain_model.commit_region(region)
-                    return brain_model
-
-                self[identifier] = LazyLoad(load)
+            self[basemodel_identifier] = LazyLoad(
+                lambda identifier=basemodel_identifier, activations_model=activations_model, layers=layers:
+                ModelCommitment(identifier=identifier, activations_model=activations_model, layers=layers))
```

Comment:
you're still referencing the `identifier` here, I think that should be `basemodel_identifier` -- or you rename `basemodel_identifier` to `identifier`

Reply:
The parameter is renamed in the lambda expression. `identifier=basemode_identifier`, so it is correct. This way I actually think it's more clear. One has to explicitly define lambda method parameters.